sketch_map_tool/upload_processing/post_process.py

```diff
@@ -0,0 +1,139 @@
+            )
+        f.geometry.coordinates = [f.geometry.coordinates[0]]  # Delete inner ring
+    return fc
+
+
+def enrich(fc: FeatureCollection, properties):
+    """Enrich GeoJSON properties and add color information to them."""
+    for feature in fc.features:
+        feature.properties = feature.properties | properties
+        if "color" in feature.properties.keys():
+            feature.properties["color"] = COLORS[feature.properties["color"]]
+    return fc
+
+
+def simplify(fc: FeatureCollection) -> FeatureCollection:
+    """Simplifies the geometries in a GeoJSON FeatureCollection.
+
+    Buffers each geometry based on a percentage of the maximum width, dissolves the
+    geometries based on the 'color' attribute, removes inner rings, and then re-applies
+    a negative buffer to restore the original size.
+
+    The function assumes that the 'color' field exists in the properties of the
+    features.
+    """
+    features = fc["features"]
+    properties = features[0]["properties"]
+    geometries = [shape(feature["geometry"]) for feature in features]
+
+    # Buffer operation
+    buffer_distance_percentage = 0.1
+    max_diag = max(
+        (
+            (geometry.bounds[2] - geometry.bounds[0]) ** 2
+            + (geometry.bounds[3] - geometry.bounds[1]) ** 2
+        )
+        ** 0.5
+        for geometry in geometries
+    )  # check for webmercator
+    buffer_distance = buffer_distance_percentage * max_diag
+    buffered_geometries = [geometry.buffer(buffer_distance) for geometry in geometries]
+    # Dissolve by color field (assuming there's a "color" field)
+    try:
+        dissolved_geometrie = [
+            remove_inner_rings(geometry)
+            for geometry in cascaded_union(buffered_geometries)
+        ]
+    except:
+        dissolved_geometrie = [
+            remove_inner_rings(geometry)
+            for geometry in [cascaded_union(buffered_geometries)]
+        ]
+
+    simplified_geometries = [
+        geometry.buffer(-buffer_distance).simplify(0.0025 * max_diag)
+        for geometry in dissolved_geometrie
+    ]
+
+    # Create a single GeoJSON feature
+    features = [
+        geojson.Feature(geometry=mapping(geometry), properties=properties)
+        for geometry in simplified_geometries
+    ]
+
+    # Create a GeoJSON feature collection with the single feature
+    fc = geojson.FeatureCollection(features)
+    return fc
+
+
+def remove_inner_rings(geometry: Polygon | MultiPolygon) -> Polygon | MultiPolygon:
+    """Removes inner rings (holes) from a given Shapely geometry object."""
+    if geometry.is_empty:
+        return geometry
+    elif geometry.type == "Polygon":
+        return Polygon(geometry.exterior)
+    elif geometry.type == "MultiPolygon":
+        return MultiPolygon([Polygon(poly.exterior) for poly in geometry.geoms])
+    else:
+        raise ValueError("Unsupported geometry type")
+
+
+def smooth(fc: FeatureCollection) -> FeatureCollection:
```

Comment:
Should smooth operate on a single shaply object instead of a feature collection. Iteration through features could happen in the "main" function.`remove_inner_rings` also operates on shaply objects.